overlord/state/task.go

```diff
@@ -159,3 +163,16 @@ func (t *Task) Get(key string, value interface{}) error {
 	t.state.ensureLocked()
 	return t.data.get(key, value)
 }
+
+// WaitFor registers another task as a requirement for t to make progress
+// and sets the status as WaitingStatus.
+func (t *Task) WaitFor(another *Task) {
+	t.state.ensureLocked()
+	t.status = WaitingStatus
+	t.waitTasks = append(t.waitTasks, another.ID())
+}
```

Comment:
Thanks, your branch looks very nice, I merged it and use it in the waitTasks now.